lib/hpke/src/certificate.cpp

```diff
@@ -193,28 +170,62 @@ struct Certificate::ParsedCertificate
 /// Certificate
 ///
 
+// Convert signature scheme for public key from EVP format to
+// to hpke variant
+static Signature::ID
+evp_sig_to_hpke_sig(EVP_PKEY* pkey)
+{
+  auto pkey_type = EVP_PKEY_id(pkey);
+  switch (pkey_type) {
+    case EVP_PKEY_ED25519:
+      return Signature::ID::Ed25519;
+    case EVP_PKEY_ED448:
+      return Signature::ID::Ed448;
+    case EVP_PKEY_EC: {
+      auto key_size = EVP_PKEY_bits(pkey);
+      return (key_size == 256) ? Signature::ID::P256_SHA256
+                               : (key_size == 384) ? Signature::ID::P384_SHA384
+                                                   : Signature::ID::P521_SHA512;
+    }
+    case EVP_PKEY_RSA:
+      return Signature::ID::RSA_SHA256;
```

Comment:
```suggestion
      // It is not necessarily the case that RSA keys are used with SHA256, 
      // but in practically all cases, it is.  Since the SubjectPublicKeyInfo
      // key doesn't tell us any more information, we assume this is the case.
      return Signature::ID::RSA_SHA256;
```